services/src/main/java/org/keycloak/services/resources/account/LinkedAccountsResource.java

```diff
@@ -0,0 +1,248 @@
+                                  EventBuilder event, 
+                                  UserModel user) {
+        this.session = session;
+        this.request = request;
+        this.client = client;
+        this.auth = auth;
+        this.event = event;
+        this.user = user;
+        realm = session.getContext().getRealm();
+    }
+    
+    @GET
+    @Path("/")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response linkedAccounts() {
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_PROFILE);
+        SortedSet<LinkedAccountRepresentation> linkedAccounts = getLinkedAccounts(this.session, this.realm, this.user);
+        return Cors.add(request, Response.ok(linkedAccounts)).auth().allowedOrigins(auth.getToken()).build();
+    }
+
+    public SortedSet<LinkedAccountRepresentation> getLinkedAccounts(KeycloakSession session, RealmModel realm, UserModel user) {
+        List<IdentityProviderModel> identityProviders = realm.getIdentityProviders();
+        SortedSet<LinkedAccountRepresentation> linkedAccounts = new TreeSet<>();
+        
+        if (identityProviders == null || identityProviders.isEmpty()) return linkedAccounts;
+        
+        Set<FederatedIdentityModel> identities = session.users().getFederatedIdentities(user, realm);
+        for (IdentityProviderModel provider : identityProviders) {
+            if (!provider.isEnabled()) {
+                continue;
+            }
+            String providerId = provider.getAlias();
+
+            FederatedIdentityModel identity = getIdentity(identities, providerId);
+
+            String displayName = KeycloakModelUtils.getIdentityProviderDisplayName(session, provider);
+            String guiOrder = provider.getConfig() != null ? provider.getConfig().get("guiOrder") : null;
+
+            LinkedAccountRepresentation rep = new LinkedAccountRepresentation();
+            rep.setConnected(identity != null);
+            rep.setProviderAlias(providerId);
+            rep.setDisplayName(displayName);
+            rep.setGuiOrder(guiOrder);
+            rep.setProviderName(provider.getAlias());
+            if (identity != null) {
+                rep.setLinkedUsername(identity.getUserName());
+            }
+            linkedAccounts.add(rep);
+        }
+        
+        return linkedAccounts;
+    }
+    
+    private FederatedIdentityModel getIdentity(Set<FederatedIdentityModel> identities, String providerId) {
+        for (FederatedIdentityModel link : identities) {
+            if (providerId.equals(link.getIdentityProvider())) {
+                return link;
+            }
+        }
+        return null;
+    }
+    
+    @GET
+    @Path("/{providerId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Deprecated
+    public Response buildLinkedAccountURI(@PathParam("providerId") String providerId, 
+                                     @QueryParam("redirectUri") String redirectUri) {
+        if (redirectUri == null) {
```

Comment:
I would add a check for manage-account here, just to prevent any abuse of this endpoint

Reply:
ok